samples/algorithms/integer-factorization/FlameGraphCounter.cs

```diff
@@ -0,0 +1,98 @@
+using System.Collections.Generic;
+using System.Diagnostics;
+
+using Microsoft.Quantum.Simulation.Core;
+using Microsoft.Quantum.Simulation.Simulators.QCTraceSimulators;
+
+namespace Microsoft.Quantum.Simulation.QCTraceSimulatorRuntime {
+    public class FlameGraphCounter : IQCTraceSimulatorListener {
+        private readonly PrimitiveOperationsCounterConfiguration configuration;
+        
+        /// <summary>
+        /// Contains all fully specified calls including resource count for the operation excluding all children.
+        /// </summary>
+        public Dictionary<string, double> FlameGraphData { get; } = new Dictionary<string, double>();
```

Comment:
Similarly here as below, this exposes a mutable data structure as readonly, such that callers can still change the contents of `FlameGraphData`; would it be worth using an ImmutableDictionary here?

Reply:
I understand the point, but would avoid the copying in this case, since the data in the dictionary can become very large.